src/storage/src/storage_state.rs

```diff
@@ -130,17 +133,16 @@ impl<'w, A: Allocate> Worker<'w, A> {
                             crate::render::build_storage_dataflow(
                                 &mut self.timely_worker,
                                 &mut self.storage_state,
-                                &source.id.to_string(),
-                                source.clone(),
+                                &id.to_string(),
+                                ingestion,
+                                as_of,
                             );
                         }
                     }
 
-                    use timely::progress::Timestamp;
-                    self.storage_state.reported_frontiers.insert(
-                        source.id,
-                        Antichain::from_elem(mz_repr::Timestamp::minimum()),
-                    );
+                    self.storage_state
+                        .reported_frontiers
+                        .insert(id, Antichain::from_elem(mz_repr::Timestamp::minimum()));
```

Comment:
wait, we do this here AND in `create_sources`??

Reply:
The one in `create_sources` is for keeping track of things in the _controller_, this one is inside a `storaged` instance that was created just for that one source. And we regularly update the state in _controller_ by sending it diffs off of the one we create here.

Reply:
That's right :+1: